auto_submit/lib/validations/revert.dart

```diff
@@ -0,0 +1,147 @@
+    }
+
+    // The requests do not match so we need to notify the user.
+    return ValidationResult(false, Action.REMOVE_LABEL,
+        'Validation of the revert request has failed. Verify the files in the revert request are the same as the original PR and resubmit the revert request.');
+  }
+
+  /// Only a team member and code owner can submit a revert request with a review.
+  bool isValidAuthor(String? author, String authorAssociation) {
+    return config.rollerAccounts.contains(author) || allowedReviewers.contains(authorAssociation);
+  }
+
+  /// The full text here is 'Reverts flutter/cocoon#XXXXX' as output by github
+  /// the link must be in the form github.com/flutter/repo/pull/id
+  final RegExp _regExp = RegExp(r'^[Rr]everts[\s]+([-\.a-zA-Z_]+/[-\.a-zA-Z_]+#[0-9]+)$', multiLine: true);
+  String? extractLinkFromText(String? bodyText) {
+    if (bodyText == null) {
+      return null;
+    }
+    var matches = _regExp.allMatches(bodyText);
+    // look at only the first match
+    if (matches.isNotEmpty) {
+      // return the first group
+      return matches.elementAt(0).group(1);
+    }
+    return null;
+  }
+
+  /// Split a reverts link on the '#' then the '/' to get the parts of the repo
+  /// slug.
+  /// It is assumed that the link has the format flutter/repo#id.
+  github.RepositorySlug _getSlugFromLink(String link) {
+    List<String> linkSplit = link.split('#');
+    List<String> slugSplit = linkSplit.elementAt(0).split('/');
+    return github.RepositorySlug(slugSplit.elementAt(0), slugSplit.elementAt(1));
+  }
+
+  /// Split a reverts link on the '#' to get the id part of the link.
+  /// It is assumed that the link has the format flutter/repo#id.
+  int _getPullRequestIdFromLink(String link) {
+    List<String> linkSplit = link.split('#');
+    return int.parse(linkSplit.elementAt(1));
+  }
+
+  /// Method to wrap functionality for getting the revert request. Done mainly
+  /// for testing purposes.
+  Future<github.PullRequest> getPullRequest(github.RepositorySlug repositorySlug, int issueId) async {
+    final GithubService gitHubService = await config.createGithubService(repositorySlug);
+    return gitHubService.getPullRequest(repositorySlug, issueId);
+  }
+
+  /// Proposer the current pull request we want to compare against proposee. If
+  /// the files in proposer are the same or a subset of proposee then we can
+  /// consider this a revert request.
+  Future<bool> comparePullRequests(
+      github.RepositorySlug repositorySlug, github.PullRequest revert, github.PullRequest current) async {
+    final GithubService githubService = await config.createGithubService(repositorySlug);
+    List<PullRequestFile> originalPullRequestFiles = await githubService.getPullRequestFiles(repositorySlug, revert);
+    List<PullRequestFile> currentPullRequestFiles = await githubService.getPullRequestFiles(repositorySlug, current);
+
+    return validateFileSetsAreEqual(originalPullRequestFiles, currentPullRequestFiles);
+  }
```

Comment:
Let's move these logics to githubService.

Reply:
Done.